placeholder-transaction-trait/src/transaction_trait.rs

```diff
@@ -0,0 +1,57 @@
+use {
+    crate::MessageTrait,
+    solana_sdk::{
+        hash::Hash,
+        packet::Packet,
+        signature::Signature,
+        transaction::{TransactionAccountLocks, TransactionError, VersionedTransaction},
+    },
+};
+
+mod reference;
+mod sanitized_transaction;
+
+pub trait TransactionTrait: MessageTrait {
+    /// Get the first signature of the message.
+    fn signature(&self) -> &Signature;
+
+    /// Get all the signatures of the message.
+    fn signatures(&self) -> &[Signature];
+
+    /// Returns the message hash.
+    // TODO: consider moving this to Message
+    fn message_hash(&self) -> &Hash;
+
+    /// Returns true if the transaction is a simple vote transaction.
+    // TODO: consider moving this to Message
+    fn is_simple_vote_transaction(&self) -> bool;
```

Comment:
same comment here as for `message_hash`

Reply:
ended up removing these 2 functions + conversion to `VersionedTransaction`.

Trimmed the traits down to what is necessary for SVM, and will have some extension that can be used inside core/runtime.